dotnet/src/Svg.cs

```diff
@@ -0,0 +1,153 @@
+﻿using Microsoft.AspNetCore.Html;
+using System;
+using System.Collections.Concurrent;
+using System.Runtime.CompilerServices;
+using System.Text.RegularExpressions;
+
+namespace StackExchange.StacksIcons;
+
+public static partial class Svg
+{
+    private const int MaxReasonableSize = 4500;
```

Comment:
Is this number just a reasonable arbitrary value or does it have more behind it? Just kinda curious.

Reply:
Honestly, no clue. It looks like this was added in 2017 with the value of `4096`. Later that year, it was changed to `4500` and has been set to that value ever since. 🤷

As Giamir mentioned, I'd love to move these checks out of the C# code and into the build step instead, so I'm not terribly worried about it right now.

Reply:
Thanks for the background! Sounds like it could be a "[cutting the ends off the ham](https://executiveforum.com/cutting-off-the-ends-of-the-ham/)" situation.

Reply:
Yes, most definitely. I think a size check is important, but I don't have a strong opinion on what size _exactly_ except "as small of a size as possible with all the tests still passing". I'll add some GH issues to track moving these checks to the build step instead of at application startup time.